datafusion/expr/src/logical_plan/builder.rs

```diff
@@ -1373,33 +1373,32 @@ pub fn union(left_plan: LogicalPlan, right_plan: LogicalPlan) -> Result<LogicalP
     }
 
     // create union schema
-    let union_qualified_fields =
+    let union_fields =
         zip(left_plan.schema().iter(), right_plan.schema().iter())
             .map(
-                |((left_qualifier, left_field), (_right_qualifier, right_field))| {
+                |((_, left_field), (_, right_field))| {
```

Comment:
There is a test that failed because of this change, and looking into the history the reason this was done was to avoid the following error:

https://github.com/apache/datafusion/issues/5410

`SchemaError(DuplicateUnqualifiedField { name: "a" })`

So it seems that this is intentional 🤔 If this only affects the Unparser, I could try stripping the qualifier only from the `Sort` node if the input node is a `Union`, but that seems a bit hacky to me.

Reply:
Further discussion that seems to indicate that removing the qualifier is the correct behavior: #2943

I'll dig into if there is a way around the `DuplicateUnqualifiedField`

Reply:
Asked on Discord why we need to enforce the `DuplicateUnqualifiedField` check - Postgres itself does not error if two columns have the same name.

Reply:
I've raised #11082 as an alternative way to do this that only affects the Unparser.